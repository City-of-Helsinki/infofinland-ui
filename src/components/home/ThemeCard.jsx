import cls from 'classnames'
import Link from 'next/link'
import Image from 'next/image'
import { longTextClass } from '@/components/Typo'

const MAX_TITLE_LENGTH = 40

const ThemeCard = ({ title, url, image, blue, green }) => (
  <div
    className={cls('rounded', {
      'bg-blue-white hover:bg-blue-lighter shadow-themecard-blue': blue,
      'bg-green-lighter hover:bg-green-light shadow-themecard-green': green,
    })}
  >
    {image && (
      <div className="hidden md:block rounded">
        <Image
          src={image}
          alt={title}
          layout="responsive"
          className="rounded-t"
        />
      </div>
    )}
    <div className="flex md:h-24">
      <p
        className={cls(
          'px-4 font-bold  my-4',
          longTextClass(title, {
            size: MAX_TITLE_LENGTH,
            classes: ['text-body-large', 'text-body'],
          })
        )}
      >
        <Link passHref href={url}>
          <a className="block">{title}</a>
        </Link>
      </p>
    </div>
  </div>
)

export default ThemeCard
