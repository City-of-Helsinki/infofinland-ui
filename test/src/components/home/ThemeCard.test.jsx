/**
 * @jest-environment jsdom
 */

import React from 'react'
// Using render and screen from test-utils.js instead of
// @testing-library/react
// import { render } from "@/test/test-utils";
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react'
import ThemeCard from '@/components/home/ThemeCard'

describe('ThemeCard', () => {
  it('should render the ThemeCard in Blue', () => {
    const { asFragment } = render(<ThemeCard title="test" url="/" blue />)
    // we can only use toBeInTheDocument because it was imported
    // in the jest.setup.js and configured in jest.config.js
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render the ThemeCard in Green', () => {
    const { asFragment } = render(<ThemeCard title="test" url="/" green />)
    // we can only use toBeInTheDocument because it was imported
    // in the jest.setup.js and configured in jest.config.js
    expect(asFragment()).toMatchSnapshot()
  })
})
