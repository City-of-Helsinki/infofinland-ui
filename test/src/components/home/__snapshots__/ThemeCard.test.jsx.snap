// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeCard should render the ThemeCard in Blue 1`] = `
<DocumentFragment>
  <a
    class="rounded block bg-blue-white hover:bg-blue-lighter shadow-themecard-blue"
    href="/"
  >
    <span
      class="flex md:h-24"
    >
      <span
        class="px-4 font-bold  my-4 text-body-large"
      >
        test
      </span>
    </span>
  </a>
</DocumentFragment>
`;

exports[`ThemeCard should render the ThemeCard in Green 1`] = `
<DocumentFragment>
  <a
    class="rounded block bg-green-lighter hover:bg-green-light shadow-themecard-green"
    href="/"
  >
    <span
      class="flex md:h-24"
    >
      <span
        class="px-4 font-bold  my-4 text-body-large"
      >
        test
      </span>
    </span>
  </a>
</DocumentFragment>
`;
